teamcity: failed tests on master: test/TestBackupRestoreInterleaved, test/TestBackupRestoreInterleaved/all_tables_in_interleave_hierarchy

The following tests appear to have failed:

[#313084](https://teamcity.cockroachdb.com/viewLog.html?buildId=313084):

```
--- FAIL: test/TestBackupRestoreInterleaved (1.610s)

------- Stdout: -------
W170802 17:09:58.763631 55274 server/server.go:298  [n?] all stores are configured as in-memory stores, so not setting up a temporary store. Queries with working set larger than memory will fail
W170802 17:09:58.763843 55274 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I170802 17:09:58.775301 55274 server/config.go:521  [n?] 1 storage engine initialized
I170802 17:09:58.775325 55274 server/config.go:523  [n?] RocksDB cache size: 512 MiB
I170802 17:09:58.775334 55274 server/config.go:523  [n?] store 0: in-memory, size 100 MiB
I170802 17:09:58.778051 55274 server/node.go:335  [n?] **** cluster 83bddf3f-48c2-4a8d-8969-48ab23cd8598 has been created
I170802 17:09:58.778075 55274 server/server.go:780  [n?] **** add additional nodes by specifying --join=127.0.0.1:32940
I170802 17:09:58.778582 55274 storage/store.go:1271  [n1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170802 17:09:58.778638 55274 server/node.go:421  [n1] initialized store [n1,s1]: {Capacity:536870912 Available:536870912 RangeCount:1 LeaseCount:0 WritesPerSecond:0}
I170802 17:09:58.778663 55274 server/node.go:319  [n1] node ID 1 initialized
I170802 17:09:58.778702 55274 gossip/gossip.go:295  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:32940" > attrs:<> locality:<> 
I170802 17:09:58.778753 55274 storage/stores.go:293  [n1] read 0 node addresses from persistent storage
I170802 17:09:58.778788 55274 server/node.go:556  [n1] connecting to gossip network to verify cluster ID...
I170802 17:09:58.778802 55274 server/node.go:581  [n1] node connected via gossip and verified as part of cluster "83bddf3f-48c2-4a8d-8969-48ab23cd8598"
I170802 17:09:58.778819 55274 server/node.go:362  [n1] node=1: started with [=] engine(s) and attributes []
I170802 17:09:58.778866 55274 sql/executor.go:423  [n1] creating distSQLPlanner with address {tcp 127.0.0.1:32940}
I170802 17:09:58.779881 55306 storage/replica_command.go:2667  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I170802 17:09:58.781398 55274 server/server.go:859  [n1] starting https server at 127.0.0.1:42029
I170802 17:09:58.781416 55274 server/server.go:860  [n1] starting grpc/postgres server at 127.0.0.1:32940
I170802 17:09:58.781427 55274 server/server.go:861  [n1] advertising CockroachDB node at 127.0.0.1:32940
W170802 17:09:58.781443 55274 sql/jobs/registry.go:148  [n1] unable to get node liveness: node not in the liveness table
W170802 17:09:58.781452 55274 sql/jobs/registry.go:273  [n1] canceling all jobs due to liveness failure
I170802 17:09:58.793702 55306 storage/replica_command.go:2667  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/tsd [r3]
I170802 17:09:58.797947 55274 sql/event_log.go:99  [n1] Event: "alter_table", target: 12, info: {TableName:eventlog Statement:ALTER TABLE system.eventlog ALTER COLUMN "uniqueID" SET DEFAULT uuid_v4() User:node MutationID:0 CascadeDroppedViews:[]}
I170802 17:09:58.799556 55306 storage/replica_command.go:2667  [split,n1,s1,r3/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r4]
I170802 17:09:58.802332 55274 sql/lease.go:341  [n1] publish: descID=12 (eventlog) version=2 mtime=2017-08-02 17:09:58.80219737 +0000 UTC
I170802 17:09:58.806497 55306 storage/replica_command.go:2667  [split,n1,s1,r4/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r5]
I170802 17:09:58.819920 55306 storage/replica_command.go:2667  [split,n1,s1,r5/1:/{Table/System���-Max}] initiating a split of this range at key /Table/11 [r6]
I170802 17:09:58.833836 55306 storage/replica_command.go:2667  [split,n1,s1,r6/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r7]
I170802 17:09:58.837083 55274 server/server.go:999  [n1] done ensuring all necessary migrations have run
I170802 17:09:58.837105 55274 server/server.go:1001  [n1] serving sql connections
I170802 17:09:58.839934 55595 sql/event_log.go:99  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:32940} Attrs: Locality:} ClusterID:83bddf3f-48c2-4a8d-8969-48ab23cd8598 StartedAt:1501693798778805027 LastUp:1501693798778805027}
I170802 17:09:58.842371 55306 storage/replica_command.go:2667  [split,n1,s1,r7/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r8]
I170802 17:09:58.849052 55306 storage/replica_command.go:2667  [split,n1,s1,r8/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r9]
I170802 17:09:58.852716 55306 storage/replica_command.go:2667  [split,n1,s1,r9/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r10]
I170802 17:09:58.859516 55306 storage/replica_command.go:2667  [split,n1,s1,r10/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r11]
I170802 17:09:58.865394 55306 storage/replica_command.go:2667  [split,n1,s1,r11/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r12]
I170802 17:09:58.872758 55306 storage/replica_command.go:2667  [split,n1,s1,r12/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r13]
I170802 17:09:58.883858 55306 storage/replica_command.go:2667  [split,n1,s1,r13/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r14]
I170802 17:09:58.917501 56022 sql/event_log.go:99  [client=127.0.0.1:49291,user=root,n1] Event: "create_database", target: 50, info: {DatabaseName:data Statement:CREATE DATABASE IF NOT EXISTS data User:root}
I170802 17:09:58.917749 55306 storage/replica_command.go:2667  [split,n1,s1,r14/1:/{Table/19-Max}] initiating a split of this range at key /Table/50 [r15]
I170802 17:09:58.922591 56022 sql/event_log.go:99  [client=127.0.0.1:49291,user=root,n1] Event: "create_table", target: 51, info: {TableName:data.bank Statement:CREATE TABLE data.bank (id INT PRIMARY KEY, balance INT, payload STRING, FAMILY (id, balance, payload)) User:root}
I170802 17:09:58.923642 55306 storage/replica_command.go:2667  [split,n1,s1,r15/1:/{Table/50-Max}] initiating a split of this range at key /Table/51 [r16]
I170802 17:09:58.926276 56121 storage/replica_command.go:2667  [n1,s1,r15/1:/{Table/50-Max}] initiating a split of this range at key /Table/51/1/1 [r17]
W170802 17:09:58.928567 56125 storage/replica.go:1978  [n1,s1,r15/1:/Table/5{0-1}] context canceled before command queue: 1 QueryTxn
I170802 17:09:58.986766 56163 storage/replica_command.go:2667  [n1,s1,r16/1:/{Table/51-Max}] initiating a split of this range at key /Table/51/1/1 [r18]
I170802 17:09:58.991239 56178 storage/replica_command.go:2667  [n1,s1,r18/1:/{Table/51/1/1-Max}] initiating a split of this range at key /Table/51/1/2 [r19]
I170802 17:09:58.995818 56203 storage/replica_command.go:2667  [n1,s1,r19/1:/{Table/51/1/2-Max}] initiating a split of this range at key /Table/51/1/3 [r20]
I170802 17:09:59.000506 56216 storage/replica_command.go:2667  [n1,s1,r20/1:/{Table/51/1/3-Max}] initiating a split of this range at key /Table/51/1/4 [r21]
I170802 17:09:59.004522 56159 storage/replica_command.go:2667  [n1,s1,r21/1:/{Table/51/1/4-Max}] initiating a split of this range at key /Table/51/1/5 [r22]
I170802 17:09:59.011931 56141 storage/replica_command.go:2667  [n1,s1,r22/1:/{Table/51/1/5-Max}] initiating a split of this range at key /Table/51/1/6 [r23]
I170802 17:09:59.016473 56294 storage/replica_command.go:2667  [n1,s1,r23/1:/{Table/51/1/6-Max}] initiating a split of this range at key /Table/51/1/7 [r24]
I170802 17:09:59.032300 56322 storage/replica_command.go:2667  [n1,s1,r24/1:/{Table/51/1/7-Max}] initiating a split of this range at key /Table/51/1/8 [r25]
I170802 17:09:59.039429 56349 storage/replica_command.go:2667  [n1,s1,r25/1:/{Table/51/1/8-Max}] initiating a split of this range at key /Table/51/1/9 [r26]
I170802 17:09:59.050120 56022 sql/event_log.go:99  [client=127.0.0.1:49291,user=root,n1] Event: "create_table", target: 52, info: {TableName:i0 Statement:CREATE TABLE i0 (a INT, b INT, PRIMARY KEY (a, b)) INTERLEAVE IN PARENT bank (a) User:root}
I170802 17:09:59.051417 55306 storage/replica_command.go:2667  [split,n1,s1,r26/1:/{Table/51/1/9-Max}] initiating a split of this range at key /Table/52 [r27]
I170802 17:09:59.059383 56022 sql/lease.go:341  [client=127.0.0.1:49291,user=root,n1] publish: descID=52 (i0) version=2 mtime=2017-08-02 17:09:59.059255132 +0000 UTC
I170802 17:09:59.067796 56022 sql/lease.go:341  [client=127.0.0.1:49291,user=root,n1] publish: descID=51 (bank) version=2 mtime=2017-08-02 17:09:59.066952842 +0000 UTC
I170802 17:09:59.085569 56022 sql/event_log.go:99  [client=127.0.0.1:49291,user=root,n1] Event: "create_table", target: 53, info: {TableName:i0_0 Statement:CREATE TABLE i0_0 (a INT, b INT, c INT, PRIMARY KEY (a, b, c)) INTERLEAVE IN PARENT i0 (a, b) User:root}
I170802 17:09:59.086361 55306 storage/replica_command.go:2667  [split,n1,s1,r27/1:/{Table/52-Max}] initiating a split of this range at key /Table/53 [r28]
I170802 17:09:59.087915 56022 sql/lease.go:341  [client=127.0.0.1:49291,user=root,n1] publish: descID=53 (i0_0) version=2 mtime=2017-08-02 17:09:59.087789782 +0000 UTC
I170802 17:09:59.096156 56022 sql/lease.go:341  [client=127.0.0.1:49291,user=root,n1] publish: descID=52 (i0) version=3 mtime=2017-08-02 17:09:59.096026414 +0000 UTC
I170802 17:09:59.111768 56022 sql/event_log.go:99  [client=127.0.0.1:49291,user=root,n1] Event: "create_table", target: 54, info: {TableName:i1 Statement:CREATE TABLE i1 (a INT, b INT, PRIMARY KEY (a, b)) INTERLEAVE IN PARENT bank (a) User:root}
I170802 17:09:59.114047 56022 sql/lease.go:341  [client=127.0.0.1:49291,user=root,n1] publish: descID=54 (i1) version=2 mtime=2017-08-02 17:09:59.113917836 +0000 UTC
I170802 17:09:59.115362 55306 storage/replica_command.go:2667  [split,n1,s1,r28/1:/{Table/53-Max}] initiating a split of this range at key /Table/54 [r29]
I170802 17:09:59.123943 56022 sql/lease.go:341  [client=127.0.0.1:49291,user=root,n1] publish: descID=51 (bank) version=3 mtime=2017-08-02 17:09:59.123582925 +0000 UTC
I170802 17:09:59.125662 56022 sql/lease.go:267  publish (count leases): descID=51 name=bank version=2 count=1
I170802 17:09:59.185403 56919 storage/replica_command.go:2667  [n1,s1,r16/1:/Table/51{-/1/1}] initiating a split of this range at key /Table/51/1/0/#/52/1/1 [r30]
I170802 17:09:59.201130 56676 storage/replica_command.go:2667  [n1,s1,r30/1:/Table/51/1/{0/#/52���-1}] initiating a split of this range at key /Table/51/1/0/#/52/1/2 [r31]
I170802 17:09:59.213517 56617 storage/replica_command.go:2667  [n1,s1,r18/1:/Table/51/1/{1-2}] initiating a split of this range at key /Table/51/1/1/#/52/1/1 [r32]
I170802 17:09:59.225356 56625 storage/replica_command.go:2667  [n1,s1,r32/1:/Table/51/1/{1/#/52���-2}] initiating a split of this range at key /Table/51/1/1/#/52/1/2 [r33]
I170802 17:09:59.230444 56877 storage/replica_command.go:2667  [n1,s1,r19/1:/Table/51/1/{2-3}] initiating a split of this range at key /Table/51/1/2/#/52/1/1 [r34]
I170802 17:09:59.235052 56689 storage/replica_command.go:2667  [n1,s1,r34/1:/Table/51/1/{2/#/52���-3}] initiating a split of this range at key /Table/51/1/2/#/52/1/2 [r35]
I170802 17:09:59.238878 57005 storage/replica_command.go:2667  [n1,s1,r20/1:/Table/51/1/{3-4}] initiating a split of this range at key /Table/51/1/3/#/52/1/1 [r36]
I170802 17:09:59.245204 56974 storage/replica_command.go:2667  [n1,s1,r36/1:/Table/51/1/{3/#/52���-4}] initiating a split of this range at key /Table/51/1/3/#/52/1/2 [r37]
I170802 17:09:59.252259 57061 storage/replica_command.go:2667  [n1,s1,r21/1:/Table/51/1/{4-5}] initiating a split of this range at key /Table/51/1/4/#/52/1/1 [r38]
I170802 17:09:59.258093 57082 storage/replica_command.go:2667  [n1,s1,r38/1:/Table/51/1/{4/#/52���-5}] initiating a split of this range at key /Table/51/1/4/#/52/1/2 [r39]
I170802 17:09:59.265971 57168 storage/replica_command.go:2667  [n1,s1,r30/1:/Table/51/1/0/#/52/1/{1-2}] initiating a split of this range at key /Table/51/1/0/#/52/1/1/#/53/1 [r40]
I170802 17:09:59.275064 57222 storage/replica_command.go:2667  [n1,s1,r31/1:/Table/51/1/{0/#/52���-1}] initiating a split of this range at key /Table/51/1/0/#/52/1/2/#/53/1 [r41]
I170802 17:09:59.281086 57238 storage/replica_command.go:2667  [n1,s1,r32/1:/Table/51/1/1/#/52/1/{1-2}] initiating a split of this range at key /Table/51/1/1/#/52/1/1/#/53/1 [r42]
I170802 17:09:59.284715 57038 storage/replica_command.go:2667  [n1,s1,r33/1:/Table/51/1/{1/#/52���-2}] initiating a split of this range at key /Table/51/1/1/#/52/1/2/#/53/1 [r43]
I170802 17:09:59.292494 57119 storage/replica_command.go:2667  [n1,s1,r34/1:/Table/51/1/2/#/52/1/{1-2}] initiating a split of this range at key /Table/51/1/2/#/52/1/1/#/53/1 [r44]
I170802 17:09:59.300841 57298 storage/replica_command.go:2667  [n1,s1,r35/1:/Table/51/1/{2/#/52���-3}] initiating a split of this range at key /Table/51/1/2/#/52/1/2/#/53/1 [r45]
I170802 17:09:59.306826 57295 storage/replica_command.go:2667  [n1,s1,r36/1:/Table/51/1/3/#/52/1/{1-2}] initiating a split of this range at key /Table/51/1/3/#/52/1/1/#/53/1 [r46]
I170802 17:09:59.313587 57151 storage/replica_command.go:2667  [n1,s1,r37/1:/Table/51/1/{3/#/52���-4}] initiating a split of this range at key /Table/51/1/3/#/52/1/2/#/53/1 [r47]
I170802 17:09:59.320918 57323 storage/replica_command.go:2667  [n1,s1,r38/1:/Table/51/1/4/#/52/1/{1-2}] initiating a split of this range at key /Table/51/1/4/#/52/1/1/#/53/1 [r48]
I170802 17:09:59.326632 57372 storage/replica_command.go:2667  [n1,s1,r39/1:/Table/51/1/{4/#/52���-5}] initiating a split of this range at key /Table/51/1/4/#/52/1/2/#/53/1 [r49]
I170802 17:09:59.348862 57444 storage/replica_command.go:2667  [n1,s1,r41/1:/Table/51/1/{0/#/52���-1}] initiating a split of this range at key /Table/51/1/0/#/54/1/1 [r50]
I170802 17:09:59.365803 57510 storage/replica_command.go:2667  [n1,s1,r50/1:/Table/51/1/{0/#/54���-1}] initiating a split of this range at key /Table/51/1/0/#/54/1/2 [r51]
I170802 17:09:59.370016 57523 storage/replica_command.go:2667  [n1,s1,r43/1:/Table/51/1/{1/#/52���-2}] initiating a split of this range at key /Table/51/1/1/#/54/1/1 [r52]
I170802 17:09:59.375791 57524 storage/replica_command.go:2667  [n1,s1,r52/1:/Table/51/1/{1/#/54���-2}] initiating a split of this range at key /Table/51/1/1/#/54/1/2 [r53]
I170802 17:09:59.380136 55611 storage/replica_command.go:2667  [n1,s1,r45/1:/Table/51/1/{2/#/52���-3}] initiating a split of this range at key /Table/51/1/2/#/54/1/1 [r54]
I170802 17:09:59.385781 57588 storage/replica_command.go:2667  [n1,s1,r54/1:/Table/51/1/{2/#/54���-3}] initiating a split of this range at key /Table/51/1/2/#/54/1/2 [r55]
I170802 17:09:59.390174 57601 storage/replica_command.go:2667  [n1,s1,r47/1:/Table/51/1/{3/#/52���-4}] initiating a split of this range at key /Table/51/1/3/#/54/1/1 [r56]
I170802 17:09:59.403294 57579 storage/replica_command.go:2667  [n1,s1,r56/1:/Table/51/1/{3/#/54���-4}] initiating a split of this range at key /Table/51/1/3/#/54/1/2 [r57]
I170802 17:09:59.412176 57659 storage/replica_command.go:2667  [n1,s1,r49/1:/Table/51/1/{4/#/52���-5}] initiating a split of this range at key /Table/51/1/4/#/54/1/1 [r58]
I170802 17:09:59.416869 57642 storage/replica_command.go:2667  [n1,s1,r58/1:/Table/51/1/{4/#/54���-5}] initiating a split of this range at key /Table/51/1/4/#/54/1/2 [r59]
I170802 17:09:59.430249 57736 ccl/storageccl/export.go:129  [n1,s1,r40/1:/Table/51/1/0/#/52/1/{1/���-2}] export [/Table/51/1/0/#/52/1/1/#/53/1,/Table/51/1/0/#/52/1/2)
I170802 17:09:59.430299 57606 ccl/storageccl/export.go:129  [n1,s1,r5/1:/Table/{SystemCon���-11}] export [/Table/4/1,/Table/4/2)
I170802 17:09:59.430667 57738 ccl/storageccl/export.go:129  [n1,s1,r31/1:/Table/51/1/0/#/52/1/2{-/���}] export [/Table/51/1/0/#/52/1/2,/Table/51/1/0/#/52/1/2/#/53/1)
I170802 17:09:59.430941 57740 ccl/storageccl/export.go:129  [n1,s1,r41/1:/Table/51/1/0/#/5{2/1/���-4/1/1}] export [/Table/51/1/0/#/52/1/2/#/53/1,/Table/51/1/0/#/54/1/1)
I170802 17:09:59.431523 57614 ccl/storageccl/export.go:129  [n1,s1,r5/1:/Table/{SystemCon���-11}] export [/Table/3/1,/Table/3/2)
I170802 17:09:59.431569 57743 ccl/storageccl/export.go:129  [n1,s1,r16/1:/Table/51{-/1/0/#/5���}] export [/Table/51/1,/Table/51/1/0/#/52/1/1)
I170802 17:09:59.431844 57744 ccl/storageccl/export.go:129  [n1,s1,r30/1:/Table/51/1/0/#/52/1/1{-/���}] export [/Table/51/1/0/#/52/1/1,/Table/51/1/0/#/52/1/1/#/53/1)
I170802 17:09:59.432133 57746 ccl/storageccl/export.go:129  [n1,s1,r50/1:/Table/51/1/0/#/54/1/{1-2}] export [/Table/51/1/0/#/54/1/1,/Table/51/1/0/#/54/1/2)
I170802 17:09:59.432532 57764 ccl/storageccl/export.go:129  [n1,s1,r51/1:/Table/51/1/{0/#/54���-1}] export [/Table/51/1/0/#/54/1/2,/Table/51/1/1)
I170802 17:09:59.432823 57766 ccl/storageccl/export.go:129  [n1,s1,r18/1:/Table/51/1/1{-/#/52/���}] export [/Table/51/1/1,/Table/51/1/1/#/52/1/1)
I170802 17:09:59.433118 57768 ccl/storageccl/export.go:129  [n1,s1,r32/1:/Table/51/1/1/#/52/1/1{-/���}] export [/Table/51/1/1/#/52/1/1,/Table/51/1/1/#/52/1/1/#/53/1)
I170802 17:09:59.433957 57774 ccl/storageccl/export.go:129  [n1,s1,r42/1:/Table/51/1/1/#/52/1/{1/���-2}] export [/Table/51/1/1/#/52/1/1/#/53/1,/Table/51/1/1/#/52/1/2)
I170802 17:09:59.434736 57782 ccl/storageccl/export.go:129  [n1,s1,r33/1:/Table/51/1/1/#/52/1/2{-/���}] export [/Table/51/1/1/#/52/1/2,/Table/51/1/1/#/52/1/2/#/53/1)
I170802 17:09:59.442054 57863 ccl/storageccl/export.go:129  [n1,s1,r43/1:/Table/51/1/1/#/5{2/1/���-4/1/1}] export [/Table/51/1/1/#/52/1/2/#/53/1,/Table/51/1/1/#/54/1/1)
I170802 17:09:59.442491 57865 ccl/storageccl/export.go:129  [n1,s1,r52/1:/Table/51/1/1/#/54/1/{1-2}] export [/Table/51/1/1/#/54/1/1,/Table/51/1/1/#/54/1/2)
I170802 17:09:59.442772 57867 ccl/storageccl/export.go:129  [n1,s1,r53/1:/Table/51/1/{1/#/54���-2}] export [/Table/51/1/1/#/54/1/2,/Table/51/1/2)
I170802 17:09:59.448070 57964 ccl/storageccl/export.go:129  [n1,s1,r19/1:/Table/51/1/2{-/#/52/���}] export [/Table/51/1/2,/Table/51/1/2/#/52/1/1)
I170802 17:09:59.449800 57966 ccl/storageccl/export.go:129  [n1,s1,r34/1:/Table/51/1/2/#/52/1/1{-/���}] export [/Table/51/1/2/#/52/1/1,/Table/51/1/2/#/52/1/1/#/53/1)
I170802 17:09:59.450110 57968 ccl/storageccl/export.go:129  [n1,s1,r44/1:/Table/51/1/2/#/52/1/{1/���-2}] export [/Table/51/1/2/#/52/1/1/#/53/1,/Table/51/1/2/#/52/1/2)
I170802 17:09:59.453610 58028 ccl/storageccl/export.go:129  [n1,s1,r35/1:/Table/51/1/2/#/52/1/2{-/���}] export [/Table/51/1/2/#/52/1/2,/Table/51/1/2/#/52/1/2/#/53/1)
I170802 17:09:59.453940 58030 ccl/storageccl/export.go:129  [n1,s1,r45/1:/Table/51/1/2/#/5{2/1/���-4/1/1}] export [/Table/51/1/2/#/52/1/2/#/53/1,/Table/51/1/2/#/54/1/1)
I170802 17:09:59.454399 58034 ccl/storageccl/export.go:129  [n1,s1,r54/1:/Table/51/1/2/#/54/1/{1-2}] export [/Table/51/1/2/#/54/1/1,/Table/51/1/2/#/54/1/2)
I170802 17:09:59.460047 58063 ccl/storageccl/export.go:129  [n1,s1,r55/1:/Table/51/1/{2/#/54���-3}] export [/Table/51/1/2/#/54/1/2,/Table/51/1/3)
I170802 17:09:59.460381 58065 ccl/storageccl/export.go:129  [n1,s1,r20/1:/Table/51/1/3{-/#/52/���}] export [/Table/51/1/3,/Table/51/1/3/#/52/1/1)
I170802 17:09:59.460664 58083 ccl/storageccl/export.go:129  [n1,s1,r36/1:/Table/51/1/3/#/52/1/1{-/���}] export [/Table/51/1/3/#/52/1/1,/Table/51/1/3/#/52/1/1/#/53/1)
I170802 17:09:59.475587 57830 ccl/storageccl/export.go:129  [n1,s1,r46/1:/Table/51/1/3/#/52/1/{1/���-2}] export [/Table/51/1/3/#/52/1/1/#/53/1,/Table/51/1/3/#/52/1/2)
I170802 17:09:59.475975 57832 ccl/storageccl/export.go:129  [n1,s1,r37/1:/Table/51/1/3/#/52/1/2{-/���}] export [/Table/51/1/3/#/52/1/2,/Table/51/1/3/#/52/1/2/#/53/1)
I170802 17:09:59.476592 57834 ccl/storageccl/export.go:129  [n1,s1,r47/1:/Table/51/1/3/#/5{2/1/���-4/1/1}] export [/Table/51/1/3/#/52/1/2/#/53/1,/Table/51/1/3/#/54/1/1)
I170802 17:09:59.491447 58161 ccl/storageccl/export.go:129  [n1,s1,r56/1:/Table/51/1/3/#/54/1/{1-2}] export [/Table/51/1/3/#/54/1/1,/Table/51/1/3/#/54/1/2)
I170802 17:09:59.492000 58229 ccl/storageccl/export.go:129  [n1,s1,r57/1:/Table/51/1/{3/#/54���-4}] export [/Table/51/1/3/#/54/1/2,/Table/51/1/4)
I170802 17:09:59.492536 58231 ccl/storageccl/export.go:129  [n1,s1,r21/1:/Table/51/1/4{-/#/52/���}] export [/Table/51/1/4,/Table/51/1/4/#/52/1/1)
I170802 17:09:59.502189 58285 ccl/storageccl/export.go:129  [n1,s1,r38/1:/Table/51/1/4/#/52/1/1{-/���}] export [/Table/51/1/4/#/52/1/1,/Table/51/1/4/#/52/1/1/#/53/1)
I170802 17:09:59.502620 58287 ccl/storageccl/export.go:129  [n1,s1,r48/1:/Table/51/1/4/#/52/1/{1/���-2}] export [/Table/51/1/4/#/52/1/1/#/53/1,/Table/51/1/4/#/52/1/2)
I170802 17:09:59.502890 58289 ccl/storageccl/export.go:129  [n1,s1,r39/1:/Table/51/1/4/#/52/1/2{-/���}] export [/Table/51/1/4/#/52/1/2,/Table/51/1/4/#/52/1/2/#/53/1)
I170802 17:09:59.510820 58340 ccl/storageccl/export.go:129  [n1,s1,r49/1:/Table/51/1/4/#/5{2/1/���-4/1/1}] export [/Table/51/1/4/#/52/1/2/#/53/1,/Table/51/1/4/#/54/1/1)
I170802 17:09:59.511220 58342 ccl/storageccl/export.go:129  [n1,s1,r58/1:/Table/51/1/4/#/54/1/{1-2}] export [/Table/51/1/4/#/54/1/1,/Table/51/1/4/#/54/1/2)
I170802 17:09:59.511564 58344 ccl/storageccl/export.go:129  [n1,s1,r59/1:/Table/51/1/{4/#/54���-5}] export [/Table/51/1/4/#/54/1/2,/Table/51/1/5)
I170802 17:09:59.515114 58373 ccl/storageccl/export.go:129  [n1,s1,r22/1:/Table/51/1/{5-6}] export [/Table/51/1/5,/Table/51/1/6)
I170802 17:09:59.515441 58375 ccl/storageccl/export.go:129  [n1,s1,r23/1:/Table/51/1/{6-7}] export [/Table/51/1/6,/Table/51/1/7)
I170802 17:09:59.515736 58377 ccl/storageccl/export.go:129  [n1,s1,r24/1:/Table/51/1/{7-8}] export [/Table/51/1/7,/Table/51/1/8)
I170802 17:09:59.519565 58452 ccl/storageccl/export.go:129  [n1,s1,r25/1:/Table/51/1/{8-9}] export [/Table/51/1/8,/Table/51/1/9)
I170802 17:09:59.519888 58454 ccl/storageccl/export.go:129  [n1,s1,r26/1:/Table/5{1/1/9-2}] export [/Table/51/1/9,/Table/51/2)
--- FAIL: test/TestBackupRestoreInterleaved/all_tables_in_interleave_hierarchy (0.450s)
sql_runner.go:80: ccl/sqlccl/backup_test.go:853: error scanning '&{0xc422c1a240 <nil>}': pq: importing 40 ranges: adding to batch: /Table/53/1/5/#/52/1/1/0/1501693799.173181125,0 -> /TUPLE/: Invalid argument: Keys must be added in order
------- Stdout: -------
W170802 17:09:59.551476 58509 server/server.go:298  [n?] all stores are configured as in-memory stores, so not setting up a temporary store. Queries with working set larger than memory will fail
W170802 17:09:59.556721 58509 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I170802 17:09:59.561906 58509 server/config.go:521  [n?] 1 storage engine initialized
I170802 17:09:59.561929 58509 server/config.go:523  [n?] RocksDB cache size: 512 MiB
I170802 17:09:59.561937 58509 server/config.go:523  [n?] store 0: in-memory, size 100 MiB
I170802 17:09:59.564295 58509 server/node.go:335  [n?] **** cluster 8163009c-1f94-4f71-b6dc-b86fd5c8da2a has been created
I170802 17:09:59.564316 58509 server/server.go:780  [n?] **** add additional nodes by specifying --join=127.0.0.1:60264
I170802 17:09:59.564792 58509 storage/store.go:1271  [n1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170802 17:09:59.564814 58509 server/node.go:421  [n1] initialized store [n1,s1]: {Capacity:536870912 Available:536870912 RangeCount:1 LeaseCount:0 WritesPerSecond:0}
I170802 17:09:59.566063 58509 server/node.go:319  [n1] node ID 1 initialized
I170802 17:09:59.566105 58509 gossip/gossip.go:295  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:60264" > attrs:<> locality:<> 
I170802 17:09:59.566515 58509 storage/stores.go:293  [n1] read 0 node addresses from persistent storage
I170802 17:09:59.566555 58509 server/node.go:556  [n1] connecting to gossip network to verify cluster ID...
I170802 17:09:59.566568 58509 server/node.go:581  [n1] node connected via gossip and verified as part of cluster "8163009c-1f94-4f71-b6dc-b86fd5c8da2a"
I170802 17:09:59.566637 58509 server/node.go:362  [n1] node=1: started with [=] engine(s) and attributes []
I170802 17:09:59.566691 58509 sql/executor.go:423  [n1] creating distSQLPlanner with address {tcp 127.0.0.1:60264}
I170802 17:09:59.574301 58509 server/server.go:859  [n1] starting https server at 127.0.0.1:47971
I170802 17:09:59.574325 58509 server/server.go:860  [n1] starting grpc/postgres server at 127.0.0.1:60264
I170802 17:09:59.574334 58509 server/server.go:861  [n1] advertising CockroachDB node at 127.0.0.1:60264
W170802 17:09:59.574362 58509 sql/jobs/registry.go:148  [n1] unable to get node liveness: node not in the liveness table
W170802 17:09:59.574373 58509 sql/jobs/registry.go:273  [n1] canceling all jobs due to liveness failure
I170802 17:09:59.579388 58509 sql/event_log.go:99  [n1] Event: "alter_table", target: 12, info: {TableName:eventlog Statement:ALTER TABLE system.eventlog ALTER COLUMN "uniqueID" SET DEFAULT uuid_v4() User:node MutationID:0 CascadeDroppedViews:[]}
I170802 17:09:59.585280 58509 sql/lease.go:341  [n1] publish: descID=12 (eventlog) version=2 mtime=2017-08-02 17:09:59.584973024 +0000 UTC
I170802 17:09:59.609447 58509 server/server.go:999  [n1] done ensuring all necessary migrations have run
I170802 17:09:59.609475 58509 server/server.go:1001  [n1] serving sql connections
I170802 17:09:59.613359 58582 storage/replica_command.go:2667  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I170802 17:09:59.617235 58920 sql/event_log.go:99  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:60264} Attrs: Locality:} ClusterID:8163009c-1f94-4f71-b6dc-b86fd5c8da2a StartedAt:1501693799566571482 LastUp:1501693799566571482}
I170802 17:09:59.626602 58582 storage/replica_command.go:2667  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/tsd [r3]
I170802 17:09:59.631491 58582 storage/replica_command.go:2667  [split,n1,s1,r3/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r4]
I170802 17:09:59.641221 58582 storage/replica_command.go:2667  [split,n1,s1,r4/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r5]
I170802 17:09:59.652619 58582 storage/replica_command.go:2667  [split,n1,s1,r5/1:/{Table/System���-Max}] initiating a split of this range at key /Table/11 [r6]
I170802 17:09:59.656295 58582 storage/replica_command.go:2667  [split,n1,s1,r6/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r7]
I170802 17:09:59.661651 58582 storage/replica_command.go:2667  [split,n1,s1,r7/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r8]
I170802 17:09:59.665532 58582 storage/replica_command.go:2667  [split,n1,s1,r8/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r9]
I170802 17:09:59.669178 58582 storage/replica_command.go:2667  [split,n1,s1,r9/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r10]
I170802 17:09:59.673005 58582 storage/replica_command.go:2667  [split,n1,s1,r10/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r11]
I170802 17:09:59.676852 58582 storage/replica_command.go:2667  [split,n1,s1,r11/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r12]
I170802 17:09:59.680375 58582 storage/replica_command.go:2667  [split,n1,s1,r12/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r13]
I170802 17:09:59.684115 58582 storage/replica_command.go:2667  [split,n1,s1,r13/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r14]
I170802 17:09:59.712578 59267 sql/event_log.go:99  [client=127.0.0.1:58131,user=root,n1] Event: "create_database", target: 50, info: {DatabaseName:ignored Statement:CREATE DATABASE ignored User:root}
I170802 17:09:59.713232 58582 storage/replica_command.go:2667  [split,n1,s1,r14/1:/{Table/19-Max}] initiating a split of this range at key /Table/50 [r15]
I170802 17:09:59.719460 59267 sql/event_log.go:99  [client=127.0.0.1:58131,user=root,n1] Event: "create_database", target: 51, info: {DatabaseName:data Statement:CREATE DATABASE data User:root}
I170802 17:09:59.721191 58582 storage/replica_command.go:2667  [split,n1,s1,r15/1:/{Table/50-Max}] initiating a split of this range at key /Table/51 [r16]
I170802 17:09:59.734164 59233 storage/replica_command.go:2667  [n1,s1,r16/1:/{Table/51-Max}] initiating a split of this range at key /Table/53/1 [r17]
I170802 17:09:59.738656 58982 storage/replica_command.go:2667  [n1,s1,r17/1:/{Table/53/1-Max}] initiating a split of this range at key /Table/53/1/0/#/54/1/1 [r18]
I170802 17:09:59.744458 59458 storage/replica_command.go:2667  [n1,s1,r18/1:/{Table/53/1/0���-Max}] initiating a split of this range at key /Table/53/1/0/#/54/1/1/#/55/1 [r19]
I170802 17:09:59.761830 59478 storage/replica_command.go:2667  [n1,s1,r19/1:/{Table/53/1/0���-Max}] initiating a split of this range at key /Table/53/1/0/#/54/1/2 [r20]
I170802 17:09:59.772724 59500 storage/replica_command.go:2667  [n1,s1,r20/1:/{Table/53/1/0���-Max}] initiating a split of this range at key /Table/53/1/0/#/54/1/2/#/55/1 [r21]
E170802 17:09:59.777813 59231 ccl/sqlccl/restore.go:735  [client=127.0.0.1:58131,user=root,n1] failed to scatter 4: kv/dist_sender.go:916: truncation resulted in empty batch on /Table/53/1/0/#/54/1/2/#/55/1{-/NULL}: AdminScatter [/Table/53/1/0/#/54/1/2/#/55/1,/Table/53/1/0/#/52/1/1)
I170802 17:09:59.778118 59574 storage/replica_command.go:2667  [n1,s1,r17/1:/Table/53/1{-/0/#/54���}] initiating a split of this range at key /Table/53/1/0/#/52/1/1 [r22]
I170802 17:09:59.782376 59348 storage/replica_command.go:2667  [n1,s1,r22/1:/Table/53/1/0/#/5{2/1/1-4/1/1}] initiating a split of this range at key /Table/53/1/0/#/52/1/2 [r23]
I170802 17:09:59.789061 59651 storage/replica_command.go:2667  [n1,s1,r21/1:/{Table/53/1/0���-Max}] initiating a split of this range at key /Table/53/1/1 [r24]
I170802 17:09:59.796056 59722 storage/replica_command.go:2667  [n1,s1,r24/1:/{Table/53/1/1-Max}] initiating a split of this range at key /Table/53/1/1/#/54/1/1 [r25]
I170802 17:09:59.808429 59737 storage/replica_command.go:2667  [n1,s1,r25/1:/{Table/53/1/1���-Max}] initiating a split of this range at key /Table/53/1/1/#/54/1/1/#/55/1 [r26]
I170802 17:09:59.816471 59816 storage/replica_command.go:2667  [n1,s1,r26/1:/{Table/53/1/1���-Max}] initiating a split of this range at key /Table/53/1/1/#/54/1/2 [r27]
I170802 17:09:59.824678 59756 storage/replica_command.go:2667  [n1,s1,r27/1:/{Table/53/1/1���-Max}] initiating a split of this range at key /Table/53/1/1/#/54/1/2/#/55/1 [r28]
E170802 17:09:59.830826 59231 ccl/sqlccl/restore.go:735  [client=127.0.0.1:58131,user=root,n1] failed to scatter 11: kv/dist_sender.go:916: truncation resulted in empty batch on /Table/53/1/1/#/54/1/2/#/55/1{-/NULL}: AdminScatter [/Table/53/1/1/#/54/1/2/#/55/1,/Table/53/1/1/#/52/1/1)
I170802 17:09:59.830919 59876 storage/replica_command.go:2667  [n1,s1,r24/1:/Table/53/1/1{-/#/54/���}] initiating a split of this range at key /Table/53/1/1/#/52/1/1 [r29]
I170802 17:09:59.835335 59916 storage/replica_command.go:2667  [n1,s1,r29/1:/Table/53/1/1/#/5{2/1/1-4/1/1}] initiating a split of this range at key /Table/53/1/1/#/52/1/2 [r30]
I170802 17:09:59.840189 59660 storage/replica_command.go:2667  [n1,s1,r28/1:/{Table/53/1/1���-Max}] initiating a split of this range at key /Table/53/1/2 [r31]
I170802 17:09:59.844776 59831 storage/replica_command.go:2667  [n1,s1,r31/1:/{Table/53/1/2-Max}] initiating a split of this range at key /Table/53/1/2/#/54/1/1 [r32]
I170802 17:09:59.848975 60013 storage/replica_command.go:2667  [n1,s1,r32/1:/{Table/53/1/2���-Max}] initiating a split of this range at key /Table/53/1/2/#/54/1/1/#/55/1 [r33]
I170802 17:09:59.857989 60056 storage/replica_command.go:2667  [n1,s1,r33/1:/{Table/53/1/2���-Max}] initiating a split of this range at key /Table/53/1/2/#/54/1/2 [r34]
I170802 17:09:59.868340 60084 storage/replica_command.go:2667  [n1,s1,r34/1:/{Table/53/1/2���-Max}] initiating a split of this range at key /Table/53/1/2/#/54/1/2/#/55/1 [r35]
E170802 17:09:59.873519 59231 ccl/sqlccl/restore.go:735  [client=127.0.0.1:58131,user=root,n1] failed to scatter 18: kv/dist_sender.go:916: truncation resulted in empty batch on /Table/53/1/2/#/54/1/2/#/55/1{-/NULL}: AdminScatter [/Table/53/1/2/#/54/1/2/#/55/1,/Table/53/1/2/#/52/1/1)
I170802 17:09:59.873620 60110 storage/replica_command.go:2667  [n1,s1,r31/1:/Table/53/1/2{-/#/54/���}] initiating a split of this range at key /Table/53/1/2/#/52/1/1 [r36]
I170802 17:09:59.884514 60151 storage/replica_command.go:2667  [n1,s1,r36/1:/Table/53/1/2/#/5{2/1/1-4/1/1}] initiating a split of this range at key /Table/53/1/2/#/52/1/2 [r37]
I170802 17:09:59.890534 60122 storage/replica_command.go:2667  [n1,s1,r35/1:/{Table/53/1/2���-Max}] initiating a split of this range at key /Table/53/1/3 [r38]
I170802 17:09:59.898815 60233 storage/replica_command.go:2667  [n1,s1,r38/1:/{Table/53/1/3-Max}] initiating a split of this range at key /Table/53/1/3/#/54/1/1 [r39]
I170802 17:09:59.909190 60279 storage/replica_command.go:2667  [n1,s1,r39/1:/{Table/53/1/3���-Max}] initiating a split of this range at key /Table/53/1/3/#/54/1/1/#/55/1 [r40]
I170802 17:09:59.913847 60325 storage/replica_command.go:2667  [n1,s1,r40/1:/{Table/53/1/3���-Max}] initiating a split of this range at key /Table/53/1/3/#/54/1/2 [r41]
I170802 17:09:59.920279 60331 storage/replica_command.go:2667  [n1,s1,r41/1:/{Table/53/1/3���-Max}] initiating a split of this range at key /Table/53/1/3/#/54/1/2/#/55/1 [r42]
E170802 17:09:59.924191 59231 ccl/sqlccl/restore.go:735  [client=127.0.0.1:58131,user=root,n1] failed to scatter 25: kv/dist_sender.go:916: truncation resulted in empty batch on /Table/53/1/3/#/54/1/2/#/55/1{-/NULL}: AdminScatter [/Table/53/1/3/#/54/1/2/#/55/1,/Table/53/1/3/#/52/1/1)
I170802 17:09:59.924299 60393 storage/replica_command.go:2667  [n1,s1,r38/1:/Table/53/1/3{-/#/54/���}] initiating a split of this range at key /Table/53/1/3/#/52/1/1 [r43]
I170802 17:09:59.928831 60353 storage/replica_command.go:2667  [n1,s1,r43/1:/Table/53/1/3/#/5{2/1/1-4/1/1}] initiating a split of this range at key /Table/53/1/3/#/52/1/2 [r44]
I170802 17:09:59.934152 60412 storage/replica_command.go:2667  [n1,s1,r42/1:/{Table/53/1/3���-Max}] initiating a split of this range at key /Table/53/1/4 [r45]
I170802 17:09:59.939157 60416 storage/replica_command.go:2667  [n1,s1,r45/1:/{Table/53/1/4-Max}] initiating a split of this range at key /Table/53/1/4/#/54/1/1 [r46]
I170802 17:09:59.947116 60207 storage/replica_command.go:2667  [n1,s1,r46/1:/{Table/53/1/4���-Max}] initiating a split of this range at key /Table/53/1/4/#/54/1/1/#/55/1 [r47]
I170802 17:09:59.952627 60571 storage/replica_command.go:2667  [n1,s1,r47/1:/{Table/53/1/4���-Max}] initiating a split of this range at key /Table/53/1/4/#/54/1/2 [r48]
I170802 17:09:59.959096 60615 storage/replica_command.go:2667  [n1,s1,r48/1:/{Table/53/1/4���-Max}] initiating a split of this range at key /Table/53/1/4/#/54/1/2/#/55/1 [r49]
E170802 17:09:59.966703 59231 ccl/sqlccl/restore.go:735  [client=127.0.0.1:58131,user=root,n1] failed to scatter 32: kv/dist_sender.go:916: truncation resulted in empty batch on /Table/53/1/4/#/54/1/2/#/55/1{-/NULL}: AdminScatter [/Table/53/1/4/#/54/1/2/#/55/1,/Table/53/1/4/#/52/1/1)
I170802 17:09:59.966823 60593 storage/replica_command.go:2667  [n1,s1,r45/1:/Table/53/1/4{-/#/54/���}] initiating a split of this range at key /Table/53/1/4/#/52/1/1 [r50]
I170802 17:09:59.971932 60678 storage/replica_command.go:2667  [n1,s1,r50/1:/Table/53/1/4/#/5{2/1/1-4/1/1}] initiating a split of this range at key /Table/53/1/4/#/52/1/2 [r51]
I170802 17:09:59.979105 60709 storage/replica_command.go:2667  [n1,s1,r49/1:/{Table/53/1/4���-Max}] initiating a split of this range at key /Table/53/1/5 [r52]
I170802 17:09:59.985524 60780 storage/replica_command.go:2667  [n1,s1,r52/1:/{Table/53/1/5-Max}] initiating a split of this range at key /Table/53/1/6 [r53]
W170802 17:09:59.985992 60695 storage/replica.go:2580  [n1,s1,r52/1:/{Table/53/1/5-Max}] context cancellation after 0.0s of attempting command [txn: c9dffe87], BeginTransaction [/Local/Range/"\xbd\x89\x8d"/RangeDescriptor,/Min), ConditionalPut [/Local/Range/"\xbd\x89\x8d"/RangeDescriptor,/Min)
I170802 17:09:59.990086 58800 vendor/google.golang.org/grpc/grpclog/grpclog.go:75  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:60264->127.0.0.1:58130: use of closed network connection

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/ccl/sqlccl/backup_test.go
+++ b/pkg/ccl/sqlccl/backup_test.go
@@ -803,8 +803,6 @@ func TestBackupRestoreResume(t *testing.T) {
 
 func TestBackupRestoreInterleaved(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	t.Skip("#17402 currently flaky")
-
 	const numAccounts = 10
 
 	_, dir, _, sqlDB, cleanupFn := backupRestoreTestSetup(t, singleNode, numAccounts)

--- a/pkg/ccl/sqlccl/restore.go
+++ b/pkg/ccl/sqlccl/restore.go
@@ -10,6 +10,7 @@ package sqlccl
 
 import (
 	"runtime"
+	"sort"
 
 	"github.com/pkg/errors"
 	"golang.org/x/net/context"
@@ -213,11 +214,22 @@ func allocateTableRewrites(
 
 	// Allocate new IDs for each table.
 	//
-	// NB: we do this in a standalone transaction, not one that covers the entire
-	// restore since restarts would be terrible (and our bulk import primitive are
-	// non-transactional), but this does mean if something fails during restore
-	// we've "leaked" the IDs, in that the generator will have been incremented.
+	// NB: we do this in a standalone transaction, not one that covers the
+	// entire restore since restarts would be terrible (and our bulk import
+	// primitive are non-transactional), but this does mean if something fails
+	// during restore we've "leaked" the IDs, in that the generator will have
+	// been incremented.
+	//
+	// NB: The ordering of the new IDs must be the same as the old ones,
+	// otherwise the keys may sort differently after they're rekeyed. We could
+	// handle this by chunking the AddSSTable calls more finely in Import, but
+	// it would be a big performance hit.
+	tables := make([]*sqlbase.TableDescriptor, 0, len(tablesByID))
 	for _, table := range tablesByID {
+		tables = append(tables, table)
+	}
+	sort.Sort(sqlbase.TableDescriptors(tables))
+	for _, table := range tables {
 		newTableID, err := sql.GenerateUniqueDescID(ctx, p.ExecCfg().DB)
 		if err != nil {
 			return nil, err

--- a/pkg/sql/sqlbase/structured.go
+++ b/pkg/sql/sqlbase/structured.go
@@ -45,6 +45,13 @@ func (ids IDs) Len() int           { return len(ids) }
 func (ids IDs) Less(i, j int) bool { return ids[i] < ids[j] }
 func (ids IDs) Swap(i, j int)      { ids[i], ids[j] = ids[j], ids[i] }
 
+// TableDescriptors is a sortable list of *TableDescriptors.
+type TableDescriptors []*TableDescriptor
+
+func (t TableDescriptors) Len() int           { return len(t) }
+func (t TableDescriptors) Less(i, j int) bool { return t[i].ID < t[j].ID }
+func (t TableDescriptors) Swap(i, j int)      { t[i], t[j] = t[j], t[i] }
+
 // ColumnID is a custom type for ColumnDescriptor IDs.
 type ColumnID parser.ColumnID
 

